src/fns/constrained_ops.nr

```diff
@@ -294,6 +294,58 @@ pub(crate) fn validate_gt<let N: u32, let MOD_BITS: u32>(lhs: [Field; N], rhs: [
     assert(result_limb == 0);
 }
 
+pub(crate) fn cmp<let N: u32, let MOD_BITS: u32>(lhs: [Field; N], rhs: [Field; N]) -> bool {
+    // so we do... p - x - r = 0 and there might be borrow flags
+    // a - b = r
+    // p + a - b - r = 0
+    let (comp_result, result, carry_flags, borrow_flags) = unsafe { __cmp_remainder(lhs, rhs) };
+    validate_in_range::<_, MOD_BITS>(result);
+
+    let borrow_shift = 0x1000000000000000000000000000000;
+    let carry_shift = 0x1000000000000000000000000000000;
+
+    let mut addend: [Field; N] = [0; N];
+    if comp_result == true {
```

Comment:
I was first thinking about skipping the if statement by casting the `bool` to `Field` and then computing `intended_lhs = (1- comp_result as Field)* lhs + comp_result * rhs` but not sure whether looping over the limbs and doing field multiplications would be any better than a conditional.

Reply:
@TomAFrench

Reply:
If you just want to swap the lhs and rhs then  you can do
```nr
let (lhs, rhs) = if comp_result {
  (rhs, lhs)
} else {
  (lhs, rhs)
}
```

This would end up with needing to merge two `[Field;N]` values but that's more efficient than duplicating the logic currently in the if-statement. (This could be made better by just returning the element-wise difference between the two arrays rather than directly reordering them)

Reply:
The best thing to do in this situation however would be to integrate the benchmarking system I added to https://github.com/noir-lang/noir-library-starter.

Reply:
thanks I was wondering if there's a way to do a swap in a single line :D